log/logger_multiwriter.go

```diff
@@ -22,36 +28,66 @@ func loggerWorker() {
 	buffer := make([]byte, 0, defaultBufferCapacity)
 	var n int
 	var err error
+
+	structuredOutbound := ExtraFields{}
 	for j := range jobsChannel {
 		if j.Passback != nil {
 			j.Passback <- struct{}{}
 			continue
 		}
-		data := j.fn()
-		buffer = append(buffer, j.Header...)
-		if j.ShowLogSystemName {
-			buffer = append(buffer, j.Spacer...)
-			buffer = append(buffer, j.SlName...)
-		}
-		buffer = append(buffer, j.Spacer...)
-		if j.TimestampFormat != "" {
-			buffer = time.Now().AppendFormat(buffer, j.TimestampFormat)
-		}
-		buffer = append(buffer, j.Spacer...)
-		buffer = append(buffer, data...)
-		if data == "" || data[len(data)-1] != '\n' {
+		msg := j.fn()
+		if j.StructuredLogging {
+			structuredOutbound[message] = msg
+			structuredOutbound[timestamp] = time.Now().UnixMilli()
```

Comment:
Aside: Not something we should here, and not part of this review, but I'm just noting where this is happening and how sub-optimal it is to have timestamping happening in a receiving jobs channel instead of log method invocation.

Reply:
Yeah you have a good point and a nice 🦅 eye, this is not exactly the best way to do things. I wanted to make sure most of the heavy lifting is completely segregated from the caller first then we can shift this back to get fields call if its needed. Prior iterations we had a major knock-on affect occurring which inhibited the caller operations, but this is not really the case anymore and if there is writer slowdown this can impact and skew the log timings.